core/src/Kokkos_Macros.hpp

```diff
@@ -356,6 +356,10 @@
 #endif
 #endif
 
+#if (1800 > KOKKOS_COMPILER_INTEL)
```

Comment:
This is OK but I find it harder to read this way than swapping the two operands (and making sure that `KOKKOS_COMPILER_INTEL` is defined).

Reply:
https://github.com/kokkos/kokkos/blob/5f8f405fa4f37422fd41972812b77973f3554d97/core/src/Kokkos_Macros.hpp#L307-L366